grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/GraknMapReduce.java

```diff
@@ -46,6 +46,8 @@
 
     private static final String RESOURCE_DATA_TYPE_KEY = "RESOURCE_DATA_TYPE_KEY";
 
+    public static final int RESERVED_TYPE_LABEL_KEY = -10;
```

Comment:
Vertex emits type label id, but has-resource edge can not. Instead, a message is sent via the edge, and a vertex property is added. So the resource vertex can emit an extra key value pair, key being this constant. Here, I just wanna choose a number that we would never use as type id. So yes, -11 is also good!